src/main/java/org/jenkinsci/plugins/github_branch_source/RepositoryUriResolver.java

```diff
@@ -38,17 +40,14 @@
 
     public static String hostnameFromApiUri(String apiUri) {
         apiUri = Util.fixEmptyAndTrim(apiUri);
-        if (apiUri == null) {
-            return "github.com";
+        try {
+            URL endpoint = new URL(apiUri);
```

Comment:
I think the idea was to use null in the field to mean GitHub public. If you want to change that, you will need a `readResolve`.

BTW it is questionable whether an HTTP(S) URL with an empty path component (as opposed to `/`) is legal, but I am not sure what the impact is of adding the trailing slash now.